pnpm-workspace.yaml

```diff
@@ -1,4 +1,5 @@
 packages:
-  - "packages/*"
-  - "examples/*"
-  - "docs"
+  - 'packages/*'
+  - 'type-export-test'
```

Comment:
`example-app-router-playground` is in fact already a good example that stress-tests all APIs of `next-intl`. Can we add a `tsconfig.declaration.json` there and add a lint task that calls `tsc` with that config?

That way we don't have to add a completely new app (each example adds some maintenance burden).

Reply:
They are used, but not everything is exported. For example `useTimeZone` hook is only used inside components and as they only export `JSX.Element`, TypeScript does not have to try and export the return type of that function. But if you add something like:
```ts
export function useExport() {
  return useTimeZone();
}
```
You'll get a following error:
```

Reply:
Awesome, thanks a lot! Just had another look from my side and added a commit with some minor cleanup.

All good from my side, let's see if CI is happy too …